QWC2Components/components/Search.jsx

```diff
@@ -304,25 +304,39 @@ class Search extends React.Component {
         );
     }
     showResult = (item, zoom=true) => {
-        if((item.type || SearchResultType.PLACE) === SearchResultType.PLACE) {
-            this.props.removeLayer("searchselection");
-            let text = item.label !== undefined ? item.label : item.text;
-            if(zoom) {
-                // find max zoom level greater than min scale
-                let maxZoom = 0;
-                const scales = this.props.map.scales;
-                for (let i in scales) {
-                    if (scales[i] < this.props.searchOptions.minScale) {
-                        break;
-                    } else {
-                        maxZoom = i;
-                    }
+        if(zoom) {
+            // find max zoom level greater than min scale
+            let maxZoom = 0;
+            const scales = this.props.map.scales;
+            for (let i in scales) {
+                if (scales[i] < this.props.searchOptions.minScale) {
+                    break;
+                } else {
+                    maxZoom = i;
                 }
+            }
 
-                // zoom to result using max zoom level
-                const newZoom = mapUtils.getZoomForExtent(CoordinatesUtils.reprojectBbox(item.bbox, item.crs, this.props.map.projection), this.props.map.resolutions, this.props.map.size, 0, maxZoom);
-                this.props.panToResult([item.x, item.y], newZoom, item.crs);
+            if(SearchResultType.THEMELAYER) {
+                item.bbox = CoordinatesUtils.reprojectBbox(item.layer.sublayers[0].bbox.bounds, item.layer.sublayers[0].bbox.crs, this.props.map.projection);
+                for (let i = 1; i < item.layer.sublayers.length; i++) {
+                    const bbox = CoordinatesUtils.reprojectBbox(item.layer.sublayers[i].bbox.bounds, item.layer.sublayers[i].bbox.crs, this.props.map.projection);
+                    item.bbox[0] = bbox[0] < item.bbox[0] ? bbox[0] : item.bbox[0];
+                    item.bbox[1] = bbox[1] < item.bbox[1] ? bbox[1] : item.bbox[1];
+                    item.bbox[2] = bbox[2] > item.bbox[2] ? bbox[2] : item.bbox[2];
+                    item.bbox[3] = bbox[3] > item.bbox[3] ? bbox[3] : item.bbox[3];
+                }
+                item.crs = this.props.map.projection;
+                item.x = 0.5 * (item.bbox[0] + item.bbox[2]);
+                item.y = 0.5 * (item.bbox[3] + item.bbox[1]);
```

Comment:
You shouldn't be editing `item`, it is part of the global state (i.e. `state.search.results`) and the state must not be mutated.